pkg/server/container_create_windows.go

```diff
@@ -68,13 +68,30 @@ func (c *criService) containerSpec(id string, sandboxID string, sandboxPid uint3
 
 	specOpts = append(specOpts, customopts.WithWindowsMounts(c.os, config, extraMounts))
 
-	specOpts = append(specOpts, customopts.WithWindowsResources(config.GetWindows().GetResources()))
-
-	username := config.GetWindows().GetSecurityContext().GetRunAsUsername()
-	if username != "" {
-		specOpts = append(specOpts, oci.WithUser(username))
+	// Start with the image config user and override below if RunAsUsername is not "".
+	username := imageConfig.User
+
+	windowsConfig := config.GetWindows()
+	if windowsConfig != nil {
+		specOpts = append(specOpts, customopts.WithWindowsResources(windowsConfig.GetResources()))
+		securityCtx := windowsConfig.GetSecurityContext()
+		if securityCtx != nil {
```

Comment:
I don't think we need all these nil checks. The `Get*` functions in protobuf should propagate nils forward, so if you do `config.GetWindows().GetSecurityContext().GetRunAsUsername()` it should return "" even if `GetWindows()` returns nil.

Reply:
Can update 👍

Reply:
which nil check is not needed they all look valid to me..

Reply:
IMO it's not a matter of will it panic.. it won't .. It's a matter of should the code proceed as if the get was successful.

Reply:
The caching of securityCtx and if securityCtx != nil could be removed in favor of just checking if `windowsConfig.GetSecurityContext.GetRunAsUsername()` and `windowsConfig.GetSecurityContext.GetCredentialSpec()` aren't empty strings. Either way works. I'm going to leave unless you feel strongly about it @kevpar

Reply:
I don't feel strongly. It seemed cleaner the other way, but agree it's a bit of an optimization to avoid doing some of the other work if an earlier `Get` call fails.